### DISCUSSION TITLE
Connection loss in Kubernetes environment

### DISCUSSION THREAD

**Discussion by rezpablo** [outline/outline#6288] [Self Hosting] [answered]
Hello,

We have Outline running on a Kubernetes cluster, we use Nginx as Ingress and we run the application with 3 replicas.

From time to time we get a 502 error message related to transport=websocket

The application remains running and responding to healthcheck, but we lose all external communication

Some things we did to try to mitigate the problem was to reduce the number of replicas to 1 and we had no effects, we also changed the proxy timeout times, but without any effects, because the problem happens without a predicted time, sometimes it takes 1 week, sometimes twice on the same day

**To Reproduce**
1. Configure a kubernetes deployment with outline with 3 replicas
2. Use an nginx-type ingress controller
3. Create the service and ingress

**Expected behavior**
No loss of connection as long as Outline remains healthy ok.

**Outline (please complete the following information):**
- Install: self hosted
- Version: 6f234be (We have had this problem since previous versions)

**Desktop (please complete the following information):**
 - OS: all
 - Browser: all
 - Version: all

**Mobile (please complete the following information):**
 - Device: all
 - OS: all
 - Browser all

**Comment by tommoor** [comment, marked as answer]
You should upgrade to v0.74.0 and see if the problem persists. A similar issue was fixed caused by a memory leak in a dependency recently.
